### PR TITLE
Update settings command

### LINKED ISSUE #17: Feature: Change user's settings

### PR DESCRIPTION
**Pull request by JTruji**
Fixes #17

### CODE REVIEW

File: src/main/scala/battery/bot/database/PersistenceService.scala [hunk 1]
```diff
@@ -1,7 +1,14 @@
...
 import battery.bot.database.DevicesQueries.{existDeviceUUID, insertDevices}
-import battery.bot.database.UsersQueries.{getUserUUID, insertUsers}
+import battery.bot.database.UsersQueries.{
+  getSettings,
+  getUserUUID,
+  insertUsers,
+  updateNightCharge,
+  updateSleepingTime,
+  updateWakeUpTime
+}
```
Comment (AdrianRaFo) [inline src/main/scala/battery/bot/database/PersistenceService.scala:11]:
```suggestion
import battery.bot.database.UsersQueries._
```

File: src/main/scala/battery/bot/core/CommandProcess.scala [hunk 2]
```diff
@@ -58,6 +59,59 @@ class CommandProcess(persistenceService: PersistenceService, telegramClient: Tel
...
+    for {
+      settingsList <- persistenceService.getUserSetting(result.message.from.username)
+      _            <- sendSettings(settingsList._1, settingsList._2, settingsList._3, result)
+    } yield ()
+  }
+
+  def updateSettings(result: Update): IO[Unit] = {
+    val data = result.message.text.split(";").toList
+    data match {
+      case comand :: sleepingTime :: wakeUpTime :: nightCharge :: Nil =>
```
Comment (AdrianRaFo) [inline src/main/scala/battery/bot/core/CommandProcess.scala:93]:
```suggestion
      case command :: sleepingTime :: wakeUpTime :: nightCharge :: Nil =>
```

File: src/main/scala/battery/bot/core/CommandProcess.scala [hunk 3]
```diff
@@ -58,6 +59,59 @@ class CommandProcess(persistenceService: PersistenceService, telegramClient: Tel
...
+        for {
+          _      <- persistenceService.updateUserSleepingTime(result.message.from.username, sleepingTime.toInt)
+          _      <- persistenceService.updateUserWakeUpTime(result.message.from.username, wakeUpTime.toInt)
+          _      <- persistenceService.updateUserNightCharge(result.message.from.username, nightCharge.toBoolean)
```
Comment (AdrianRaFo) [inline src/main/scala/battery/bot/core/CommandProcess.scala:97]: Shouldn't we just update all settings in the same query?

Comment (JTruji) [reply src/main/scala/battery/bot/core/CommandProcess.scala]: Queries joined

File: src/main/scala/battery/bot/core/CommandProcess.scala [hunk 4]
```diff
@@ -58,6 +58,61 @@ class CommandProcess(persistenceService: PersistenceService, telegramClient: Tel
...
+  def updateSettings(result: Update): IO[Unit] = {
+    val data = result.message.text.split(";").toList
+    data match {
+      case _ :: sleepingTime :: wakeUpTime :: nightCharge :: Nil =>
+        for {
+          _ <- persistenceService.updateUserSettings(
+            result.message.from.username,
+            sleepingTime.toInt,
+            wakeUpTime.toInt,
+            nightCharge.toBoolean
```
Comment (AdrianRaFo) [inline src/main/scala/battery/bot/core/CommandProcess.scala:97]: If any of these conversions fails we should capture the errors and send an error response to the user

**Review by AdrianRaFo — APPROVED** [review]

**Pull request merged**